proto/internal/temporal/server/api/deployment/v1/message.proto

```diff
@@ -111,6 +111,9 @@ message VersionLocalState {
     // Key: Task Queue Name
     map<string, TaskQueueFamilyData> task_queue_families = 10;
 
+    // Number of task queues which will be synced in a single batch.
+    int32 sync_batch_size = 11;
```

Comment:
Why does this value have to be persisted in version or deployment state at all? It's only ever overridden with a test hook, right? So just have a global function to get the value (from a test hook or default) where it's needed

Reply:
I haven't worked with our worker workflows too much, but I assumed that the testhook cannot be injected there? The first version of the PR suggested a global variable, so I suggested this direction instead. If the testhook is (safely) accessible from a worker workflow, then that would indeed be easier.

Reply:
I see that there's an `fx.go` for worker deployment. @Shivs11 could you obtain the testhook instance from there so you don't have to pass it via proto?